### PR TITLE
WebGLRenderLists: Use stack approach.

### PR DESCRIPTION
**Pull request by Mugen87**
Related issue: see https://github.com/mrdoob/three.js/pull/9545#issuecomment-777420260

**Description**

Using the same stack approach like with render states.

### CODE REVIEW

File: src/renderers/webgl/WebGLCubeMaps.js [hunk 1]
```diff
@@ -40,15 +40,13 @@ function WebGLCubeMaps( renderer ) {
...
 
-						const currentRenderList = renderer.getRenderList();
 						const currentRenderTarget = renderer.getRenderTarget();
 
 						const renderTarget = new WebGLCubeRenderTarget( image.height / 2 );
 						renderTarget.fromEquirectangularTexture( renderer, texture );
 						cubemaps.set( texture, renderTarget );
 
 						renderer.setRenderTarget( currentRenderTarget );
-						renderer.setRenderList( currentRenderList );
```
Comment (Mugen87) [inline src/renderers/webgl/WebGLCubeMaps.js:51]: The good thing is that these calls are not necessary anymore^^.

Comment (mrdoob) [reply src/renderers/webgl/WebGLCubeMaps.js:51]: Good riddance! 😬👌

**Comment by mrdoob** [comment]
Thanks!

**Comment by mrdoob** [comment]
FYI @Oletus

File: src/renderers/webgl/WebGLRenderLists.js [hunk 2]
```diff
@@ -174,24 +174,26 @@ function WebGLRenderLists( properties ) {
...
 		let list;
 
-		if ( cameras === undefined ) {
+		if ( lists.has( scene ) === false ) {
 
 			list = new WebGLRenderList( properties );
-			lists.set( scene, new WeakMap() );
-			lists.get( scene ).set( camera, list );
+			lists.set( scene, [] );
+			lists.get( scene ).push( list );
```
Comment (mrdoob) [inline src/renderers/webgl/WebGLRenderLists.js:185]: I think we can just do this?

```js
lists.set( scene, [ list ] );
```

Comment (mrdoob) [reply src/renderers/webgl/WebGLRenderLists.js:185]: I'll change it locally.

Comment (Mugen87) [reply src/renderers/webgl/WebGLRenderLists.js:185]: The same change can be applied in `WebGLRenderStates`, too.

Comment (Mugen87) [reply src/renderers/webgl/WebGLRenderLists.js:185]: Sorry, you were faster than my comment 😅  af07e8ad159a7106da5d2984813e3a282a8e4b75

**Pull request merged**